README.md

```diff
@@ -1 +1,94 @@
+
+## Plugin Utils
+Common optional utilities to be used when developing kaiju plugins. There is a dependency on webpack for preview creation. If you don't need webpack, you might not want to use these utils.
+
+### Usage
+```js
+const { PluginUtils } = require('kaiju-plugin-utils');
+
+PluginUtils.rootPath() // for example
+```
+
+### runCompiler
+Runs the webpack compiler on the files included in the memory file system with the config supplied.
+
+**Arguments:**
+Memory File System - A memory file system containing the files to be webpacked.
+
+Config - The webpack config to be used by the compiler.
+
+**Return values:** Compiler file system - A promise for a memory file system contained the output from the webpack compiler.
+
+### webpackCompiler
+Creates a webpack compiler setup with the config and memory file system passed in and a new memory file system instance for the output.
+
+**Arguments:**
+
+Memory File System - A memory file system containing the files to be webpacked.
+
+Config - The webpack config to be used by the compiler.
+
+**Return values:** A webpack compiler setup with a memory file system.
+
+### defaultWebpackConfig
+A basic optional webpack config. Importantly it setups the resolve to point to the root app's node modules, output based on arguments and the output and it sets the environment to production.
+
+**Arguments:**
+public path - the path the the preview's assets, this will be supplied by the kaiju server.
+
+output path - the location to build the assets to, this should be returned to the kaiju server.
+
+**Return values:** A webpack config object.
+
+### webpackFs
+A setup fallback file system proxy with the passed in memory fs and a standard filesystem.
+
+**Arguments:**
+
+MemoryFS - a memory file system instance.
+
+**Return values:** A fallback file system.
+
+### rootPath
+The path the the root of the kaiju node server. Uses the app-root-path library.
+
+**Return values:** A string representing the root path.
+
+## Fallback File System
+The fallback file system is a very specific proxy setup on top of a memory file system. This filesystem falls back to the supplied file system when a file isn't found in the memory file system. This was created to allow webpack to use the kaiju servers node modules instead of requiring an npm install on the in memory file system to generate the preview. The fs methods that will 'fallback' are readFileSync, statSync, readlinkSync, existsSync and readdirSync. These were chosen based on webpacks usage.
+
+### Usage
+```js
+const {, fallbackFs} = require('kaiju-plugin-utils');
```

Comment:
Extra comma

Reply:
The extra comma is intended, we're using destructuring to include just fallbackFs.

Reply:
I was wrong (for the better), will be fixing.